packages/powersync_attachments_stream/lib/src/abstractions/attachment_context.dart

```diff
@@ -0,0 +1,49 @@
+import '../attachment.dart';
+
+/// Context for performing Attachment operations.
+///
+/// This is typically provided through a locking/exclusivity method and allows
+/// safe, transactional operations on the attachment queue.
+abstract class AbstractAttachmentContext {
+  /// Delete the attachment from the attachment queue.
+  ///
+  /// [id]: The ID of the attachment to delete.
+  /// [tx]: The database context to use for the operation.
+  Future<void> deleteAttachment(String id, dynamic context);
```

Comment:
Also this function isn't called at all, I assume that should happen in `SyncingService.deleteAttachment`?